### PR TITLE
Subscribe Unsupported Donors

### PR DESCRIPTION
**Pull request by seanpdoyle**
https://trello.com/c/nhioUOIz

When a potential donor is in a not-yet-launched zipcode, allow them to
sign up to be notified once we launch in their area.

This commit introduces a `Subscription` model which represents a pairing
of emails and zipcodes.

### CODE REVIEW

File: app/controllers/subscriptions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    end
+
+    @subscription = Subscription.new(zipcode: zipcode)
+  end
+
+  def create
+    @subscription = build_subscription
+
+    if @subscription.save
+      redirect_to page_url(:thanks)
```
Comment (croaky) [inline app/controllers/subscriptions_controller.rb]: Didn't realize `page_url` could take a symbol; that's nice.

File: app/models/subscription.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class Subscription < ActiveRecord::Base
+  validates :email, presence: true, uniqueness: { scope: :zipcode }
```
Comment (croaky) [inline app/models/subscription.rb]: EmailValidator?

File: app/views/subscriptions/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<%= simple_form_for subscription do |f| %>
+  <%= f.input :email %>
```
Comment (croaky) [inline app/views/subscriptions/_form.html.erb]: Simple Form turns this into an HTML5 email input, right?

File: config/routes.rb [hunk 4]
```diff
@@ -4,6 +4,8 @@
 
   resources :registrations, only: [:create, :new]
   resources :pre_registrations, only: [:create]
+  resources :subscriptions, only: [:create, :new]
```
Comment (croaky) [inline config/routes.rb]: `:sort`?

File: db/migrate/20160331234248_create_subscriptions.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class CreateSubscriptions < ActiveRecord::Migration
+  def change
+    create_table :subscriptions do |t|
+      t.string :email, null: false
+      t.string :zipcode, null: false
```
Comment (croaky) [inline db/migrate/20160331234248_create_subscriptions.rb]: Timestamps?

**Comment by croaky** [comment]
Looks good.

File: app/views/subscriptions/_form.html.erb (hunk 3 above)
Comment (seanpdoyle) [inline app/views/subscriptions/_form.html.erb]: screen shot 2016-04-01 at 9 15 25 am

@croaky this was news to me. Nifty!

File: db/migrate/20160331234248_create_subscriptions.rb (hunk 5 above)
Comment (seanpdoyle) [inline db/migrate/20160331234248_create_subscriptions.rb]: fixed!

**Pull request merged**